docs/proposals/network/README.md

```diff
@@ -0,0 +1,116 @@
+# Software Defined Network
+
+## Analysis
+
+### Examples
+
+- Flannel: good example, focused on networking
+- Calico
+- Weave
+- Cilium
+
+### Sources
+
+- [CNI](https://github.com/containernetworking/cni) also make a good framework for creating a new container networking project from scratch
+- [Networks - treeptik k8s](https://treeptik.gitbook.io/k8s/fundamentals/)
+- [Explain NS linux](https://www.youtube.com/watch?v=j_UUnlVC2Ss)
+- [Spec - CNI](https://www.cni.dev/docs/spec/)
+
+## Network domain
+
+### Vocabulary
+
+- *runtime* is the program responsible for executing CNI plugins
+- *plugin* is a program that applies a specified network configuration
+
+### Definition
+
+The *plugins* (= binaries) needs to be installed on every node in the cluster.
+
+For that, we have to create *plugins* (= binaries) implementing the CNI specifications.
+
+That way, the *runtime* (e.g. the node agent) can use the *plugins* to create network interfaces between pods.
+
+The goal is to be able to:
+
+- manage networks through the software
+- isolate instance/pods with isolated networks
+- apply network security policies
+
+### Scope
+
+- Pod-to-Pod network packets on the same host
+	- Firewall (e.g. iptables)
+		- Need to ACCEPT FORWARDING traffic for pod CIDR because packets are dropped by the Linux kernel by default  
+		  (Linux treats network packets in non-default network namespaces as external packets by default)
+	- Allow outgoing internet
+- Cross-node pod-to-pod network packets
+- Port mapping / Selectors pods
+- DNS routing (e.g. ingress controller)
+- DNS resolution (e.g. coreDNS) -> pod-to-pod and pod-to-service
+	- "<NAME_SVC>.<NAMESPACE>.svc.cluster.local"
+	- "<POD_IP_WITH_DASH_INSTEAD_DOT>.<NAMESPACE>.pod.cluster.local"
+
+### Out of scope
+
+| Not for now                                      | Not for us |
+|--------------------------------------------------|------------|
+| RBAC, Network Policies                           | kube proxy |
+| Monitoring network traffic (e.g. metrics-server) |            |
+| Load balancing (e.g. service)                    |            | 
+| Load balancer (e.g. klipper, metallb)            |            |
+
+### Components list of a minimal k8s (sample)
+
+- **CNI plugins** (us)
+- etcd (not us)
+- scheduler (not us)
+- apiserver (not us)
+- controller (not us)
+
+## API - Component interface
+
+### CNI
```

Comment:
So you're planning to implement a CNI-compliant plugin? That's fine with me, but this section should be very explicit about it. Something like:

```
The `orka` SDN implements the [`CNI`](https://www.cni.dev/docs/spec/) specification. 
It is a `CNI` plugin and must be called by e.g. the node Agent through the `CNI` specified
command line interface. It configure workloads networking according to a set of `CNI`
compliant configuration files, as described in the following sections.


```

Reply:
We did say that in the [Definition section](https://github.com/dev-sys-do/orka/blob/57dc5fc4101f50b1bfdad128af6968fdcda90ec2/docs/proposals/network/README.md#definition), but maybe it was not clear enough about the link to orka SDN.

I've used your paragraph and modified it slightly to update this section ([Definition](https://github.com/dev-sys-do/orka/blob/57dc5fc4101f50b1bfdad128af6968fdcda90ec2/docs/proposals/network/README.md#definition)).

I don't know if we should explain the goal of the SDN in the API section (instead of the Definition section), I prefer the opposite to reserve the API section with more explicit explanation?

Reply:
Yes, this is fine to me. It's clearer now, thanks.